include/os/freebsd/spl/sys/zone.h

```diff
@@ -29,19 +29,18 @@
 #ifndef _OPENSOLARIS_SYS_ZONE_H_
 #define	_OPENSOLARIS_SYS_ZONE_H_
 
+#include <sys/jail.h>
+
 /*
  * Macros to help with zone visibility restrictions.
  */
 
-#define	GLOBAL_ZONEID	0
+#define	GLOBAL_ZONEID	(&prison0)
 
 /*
- * Is thread in the global zone?
+ * Is proc in the global zone?
  */
-#define	INGLOBALZONE(p)	in_globalzone((p))
-
-
-extern boolean_t in_globalzone(struct proc *);
+#define	INGLOBALZONE(p)	(!jailed(FIRST_THREAD_IN_PROC((p))->td_ucred))
```

Comment:
Why not just (!jailed((p)->p_ucred)) ?  I see one such example in kernel already.